### PR TITLE
XD-3385 Added CloudConfiguration via @Import for "cloud" profiles

### PR DESCRIPTION
**Pull request by trisberg**
- this allows us to handle "lattice" and "!lattice" profiles only for "cloud"

### CODE REVIEW

File: spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/config/AdminConfiguration.java [hunk 1]
```diff
@@ -87,27 +86,9 @@ public ModuleDeployer moduleDeployer(ModuleLauncher moduleLauncher) {
 
 	@Configuration
 	@Profile("cloud")
+	@Import(CloudConfiguration.class)
```
Comment (markfisher) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/config/AdminConfiguration.java]: would this work if the `@Import` were at class level in AdminApplication, and then `@Profile` were at class level for CloudConfiguration?

Comment (trisberg) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/config/AdminConfiguration.java]: looks like it would, I'll re-work this

**Comment by trisberg** [comment]
replaced the earlier commit with a new one - 8ee05ff

File: spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/config/CloudConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.context.annotation.Profile;
+
+
+/**
+ * Configuration for cloud profiles (cloud, lattice)
+ *
+ * @author Thomas Risberg
+ */
+@Profile("cloud")
+public class CloudConfiguration {
```
Comment (jvalkeal) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/config/CloudConfiguration.java]: So either one of these beans, `ReceptorModuleDeployer` or `CloudFoundryModuleDeployer` is always created, right? Then we will get into same dance with expecting other configs to work in a correct order to override these? I assume this is how `LocalModuleDeployer` would then get created. I don't really see no reliable way getting my yarn deployer configured via this profile structure.

Comment (jvalkeal) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/config/CloudConfiguration.java]: Ah there's `@Profile("cloud")` above but still this feels brittle.

Comment (markfisher) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/config/CloudConfiguration.java]: I think it's better (at least more reliable) than what we had before, so I will merge it. We can still consider other options.

**Comment by markfisher** [comment]
squashed and merged as https://github.com/spring-cloud/spring-cloud-data/commit/8ef07a1a55edd0c42235432f06b18cd0d6de55c5

**Pull request closed without merging**